doc/code_data_relocation/relocation.rst

```diff
@@ -0,0 +1,97 @@
+.. _code_data_relocation:
```

Comment:
As noted elsewhere, this document should go into the developer guide area, so move this from ``doc/code_data_relocation/relocation.rst`` to ``doc/application/relocation.rst`` and edit the ``doc/application/index.rst`` to include ``relocation.rst`` into the toc there.